DataFormats/PatCandidates/interface/PATObject.h

```diff
@@ -333,7 +329,7 @@ namespace pat {
     /// Get human-readable type of user data object, for debugging
     const std::string &userDataObjectType(const std::string &key) const {
       const pat::UserData *data = userDataObject_(key);
-      return (data != nullptr ? data->typeName() : pat_statics::EMPTY_STR);
+      return (data != nullptr ? data->typeName() : "");
```

Comment:
This is not correct. Using `""` means the compiler will create a temporary `std::string` and return the address to that and then immediately delete that temporary at the end of that function call.

Reply:
You could do the same thing you did for `get_empty_cpv` and create a `get_empty_string` function.

Reply:
Doh, good point...